src/ggl-setup.py

```diff
@@ -0,0 +1,524 @@
+
+
+def _stop_and_disable_services():
+    commands = [
+        ['sudo', 'systemctl', 'stop', 'greengrass-lite.target'],
+        ['sudo', 'systemctl', 'disable', 'greengrass-lite.target'],
+        ['sudo', 'systemctl', 'disable', 'ggl.aws_iot_tes.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.aws_iot_mqtt.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_config.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_health.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_fleet_status.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_deployment.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_pubsub.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg-ipc.socket.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.ggconfigd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.iotcored.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.tesd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.ggdeploymentd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.gg-fleet-statusd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.ggpubsubd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.gghealthd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.ggipcd.service'],
+        ['sudo', 'systemctl', 'daemon-reload'],
+        ['sudo', 'rm', '-rf', '/etc/systemd/system/ggl*']
+    ]
+
+    for cmd in commands:
+        try:
+            subprocess.run(cmd, check=True)
+        except Exception as e:
+            print(f"Error when executing {' '.join(cmd)}: {e}")
+            return False
+    print("Successfully stopped and disabled all sevices")
+    return True
+
+
+# ===============================================
+# FILE & DIR FUNCTIONS
+# ===============================================
+
+
+def _create_dir(dir: str,
+                permission=False,
+                octal=None,
+                ownership=False,
+                flag=None,
+                user=None,
+                group=None) -> bool:
+    try:
+        if not os.path.exists(dir):
+            subprocess.run(['sudo', 'mkdir', dir], check=True)
+        if permission and octal is not None:
+            subprocess.run(['sudo', 'chmod', octal, dir], check=True)
+        if ownership and user is not None and group is not None:
+            if flag is not None:
+                subprocess.run(['sudo', 'chown', flag, f'{user}:{group}', dir],
+                               check=True)
+            else:
+                subprocess.run(['sudo', 'chown', f'{user}:{group}', dir],
+                               check=True)
+        return True
+    except subprocess.CalledProcessError as e:
+        print(f"Error when creating directory: {str(e)}")
+        return False
+
+
+def _remove_dir(dir: str) -> bool:
+    try:
+        if os.path.exists(dir):
+            subprocess.run(['sudo', 'rm', '-rf', dir], check=True)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method is running with elevated privileges. Privilege elevation occurs when you access resources with an extended set of permissions. When unattended, these privileges might allow attackers to compromise the system. Do not use elevated privileges when they can be avoided.

[Learn more](https://cwe.mitre.org/data/definitions/269.html)